core/jvm/src/main/scala/zio/sql/expr.scala

```diff
@@ -108,7 +108,21 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
-    implicit def literal[A: TypeTag](a: A): Expr[Features.Literal, Any, A] = Expr.Literal(a)
+    implicit def literal[A](a: A)(implicit typeTag: TypeTag[A]): Expr[Features.Literal, Any, A] =
+      typeTag match {
+        case TypeTag.TBoolean   => Expr.Literal(a)
+        case TypeTag.TByte      => Expr.Literal(a)
+        case TypeTag.TByteArray => Expr.Literal(a) // TODO: not sure about this one
+        case TypeTag.TDouble    => Expr.Literal(a)
+        case TypeTag.TFloat     => Expr.Literal(a)
+        case TypeTag.TInt       => Expr.Literal(a)
+        case TypeTag.TLong      => Expr.Literal(a)
+        case TypeTag.TShort     => Expr.Literal(a)
+        case _                  => literalAsString(a)
+      }
+
+    private def literalAsString[A](a: A) =
+      Expr.Literal(s"'${a.toString}'").asInstanceOf[Expr[Features.Literal, Any, A]]
```

Comment:
Generally it's a good direction, but I have the same doubts about it as in https://github.com/zio/zio-sql/pull/269.
This really fits better into rendering.

Reply:
So TBH I would drop this change and make the test wrap the String params in single quotes for the moment. Ugly I know. But it's not a precedent. Someone will fix this in another PR in the renderer.

Reply:
I can change it in the afternoon (together with conflict resolving), but I will need to add some nasty workaround to tests added to `PostgresModuleTest.scala`.